src/kitovu/sync/syncplugin.py

```diff
@@ -0,0 +1,46 @@
+"""Loading and handling of sychronization plugins."""
+
+import abc
+import pathlib
+import typing
+
+
+class AbstractSyncPlugin(metaclass=abc.ABCMeta):
+
+    """The specification/"interface" a synchronization plugin implements.
+
+    Every abstract method in this class is a plugin hook.
+    """
+
+    @abc.abstractmethod
+    def connect(self, url: str, options: dict) -> None:
+        """Connect to the given URL.
+
+        This should raise an appropriate exception if the connection failed.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def disconnect(self) -> None:
+        """Close any open connection."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
```

Comment:
chicken/egg question: we define here two functions, create_local_digest, create_remote_digest, but the smb-implementation adds another function, create_digest, that is used by both functions which were implemented from this abstract class. Is this a wise decision (code repetition)? shouldn't we better define this differently, e.g. with a create_digest-function in this class as well?

Reply:
Every plugin can decide for itself how a digest would look - e.g. for SMB we use the size and modification time, but with e.g. moodle it might be possible to use some kind of hash we can get from the server. This means every plugin needs to have a `create_local_digest` and a `create_remote_digest`, because the local digest looks different based on the plugin.

The `_create_digest` in `smb.py` is just a (private) utility function so that part of the code *isn't* duplicated in `smb.py`.